Makefile

```diff
@@ -0,0 +1,270 @@
+-------------------------------------
+
+1. install dependencies in the current directory with:
+make install
+
+2. initialize the database with:
+make db-init
+
+3. initialize the Django database and set a admin (superuser) account with:
+make django-init
+
+4. to run Riodgeback, first start Postgres, RabbitMQ, and Celery with:
+make start-services
+
+5. start the main Django development server with:
+make runserver
+
+Extras
+------
+
+Shutdown all services:
+make stop-services
+
+Check if pre-configured ports are already occupied on your system:
+make check-port-collision
+
+Consult the contents of this Makefile for other management recipes.
+
+endef
+export help
+help:
+	@printf "$$help"
+
+.PHONY : help
+
+
+# ~~~~~ Python and nextflow installation ~~~~~~ #
+PATH:=$(CURDIR)/conda/bin:$(PATH)
+unexport PYTHONPATH
+unexport PYTHONHOME
+
+ifeq ($(UNAME), Darwin)
+CONDASH:=Miniconda3-4.7.12.1-MacOSX-x86_64.sh
+endif
+
+ifeq ($(UNAME), Linux)
+CONDASH:=Miniconda3-4.7.12.1-Linux-x86_64.sh
+endif
+
+CONDAURL:=https://repo.continuum.io/miniconda/$(CONDASH)
+
+conda:
+	@echo ">>> Setting up conda..."
+	@wget "$(CONDAURL)" && \
+	bash "$(CONDASH)" -b -p conda && \
+	rm -f "$(CONDASH)"
+
+# NOTE: RabbitMQ sometimes has installation problems on macOS 10.12
+install: conda
+	conda install -y \
+	anaconda::postgresql \
+	conda-forge::ncurses \
+	rabbitmq-server && \
+	pip install -r requirements.txt
+
+# Ridgeback environment variables for configuration
+export RIDGEBACK_DB_NAME:=db
+export RIDGEBACK_DB_USERNAME:=$(shell whoami)
+export RIDGEBACK_DB_PASSWORD:=admin
+export RIDGEBACK_DB_URL:=localhost
+export RIDGEBACK_PORT:=1111
+export RIDGEBACK_TOIL_DIR:=$(CURDIR)/toil
+export RIDGEBACK_TOIL_JOB_STORE:=$(RIDGEBACK_TOIL_DIR)/job_store
+export RIDGEBACK_TOIL_JOB_STORE_ROOT:=$(RIDGEBACK_TOIL_DIR)/job_store_root
+export RIDGEBACK_TOIL_WORK_DIR_ROOT:=$(RIDGEBACK_TOIL_DIR)/work
+export RIDGEBACK_TOIL_TMP_DIR_ROOT:=$(RIDGEBACK_TOIL_DIR)/tmp
+export RIDGEBACK_LSF_WALLTIME:=100
+export RIDGEBACK_LSF_SLA:=CMOPI
+
+$(RIDGEBACK_TOIL_DIR):
+	mkdir -p "$(RIDGEBACK_TOIL_DIR)"
+$(RIDGEBACK_TOIL_JOB_STORE):
+	mkdir -p "$(RIDGEBACK_TOIL_JOB_STORE)"
+$(RIDGEBACK_TOIL_JOB_STORE_ROOT):
+	mkdir -p "$(RIDGEBACK_TOIL_JOB_STORE_ROOT)"
+$(RIDGEBACK_TOIL_WORK_DIR_ROOT):
+	mkdir -p "$(RIDGEBACK_TOIL_WORK_DIR_ROOT)"
+$(RIDGEBACK_TOIL_TMP_DIR_ROOT):
+	mkdir -p "$(RIDGEBACK_TOIL_TMP_DIR_ROOT)"
+
+toil-init: $(RIDGEBACK_TOIL_DIR) $(RIDGEBACK_TOIL_JOB_STORE) $(RIDGEBACK_TOIL_JOB_STORE_ROOT) $(RIDGEBACK_TOIL_WORK_DIR_ROOT) $(RIDGEBACK_TOIL_TMP_DIR_ROOT)
+
+# ~~~~~ Postgres Databse Setup ~~~~~ #
+export PGDATA:=$(RIDGEBACK_DB_NAME)
+export PGUSER:=$(RIDGEBACK_DB_USERNAME)
+export PGPASSWORD:=$(RIDGEBACK_DB_PASSWORD)
+export PGHOST:=$(RIDGEBACK_DB_URL)
+export PGPORT:=$(RIDGEBACK_PORT)
+export PGLOG:=$(LOG_DIR_ABS)/postgres.log
+export PGDATABASE:=$(RIDGEBACK_DB_NAME)
+
+$(PGDATA):
+	mkdir -p "$(PGDATA)"
+db-start:
+	pg_ctl -D "$(PGDATA)" -l "$(PGLOG)" start
```

Comment:
I think we should update to the following, otherwise it defaults to 5432
```
 	pg_ctl -D "$(PGDATA)" -o "-p $(PGPORT)" -l "$(PGLOG)" start
```

Reply:
Postgres is supposed to be resolving the Port from the `PGPORT` environment variable, which is inherited from `RIDGEBACK_PORT`, both of which can be overridden in the cli invocation with `make db-start RIDGEBACK_PORT=<integer>`. Is that not the behavior you are seeing?

Reply:
I wonder if it's a discrepency with the Postgres version. I see PGPORT in the docs for `postgres` in 11, but not for `pg_ctl`. In PostgreSQL 8 it's documented for `pg_ctl`. Just something to note.